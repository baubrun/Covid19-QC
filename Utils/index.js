import checkPropTypes from "check-prop-types";


export const findByTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test="${attr}"]`);
  return wrapper;
};

export const checkProps = (component, expected) => {
  const propsError = checkPropTypes(
    component.propTypes,
    expected,
    "props",
    component.name
  );
  return propsError
};
