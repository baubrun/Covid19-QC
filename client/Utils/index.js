import checkPropTypes from "check-prop-types";


export const checkProps = (component, expected) => {
  const propsError = checkPropTypes(
    component.propTypes,
    expected,
    "props",
    component.name
  );
  return propsError
};
