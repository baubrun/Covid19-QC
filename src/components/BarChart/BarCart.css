.barCart-container {
    position: relative;
    height: 30vh; 
    width: 70vw;
}