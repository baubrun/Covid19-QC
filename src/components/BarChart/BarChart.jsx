import React from "react";
import { Bar } from "react-chartjs-2";

const BarChart = ({ data, region }) => {
  const cases = data.find((d) => d.région === region);

  return (
    <>
      {!region ? (
        ""
      ) : (
        <Bar
          data={{
            labels: ["Confirmés", "Décès"],
            datasets: [
              {
                label: "People",
                backgroundColor: [
                  "rgba(129, 233, 121, 0.5)",
                  "rgba(249, 87, 56, 0.6)",
                ],
                data: [cases.confirmés, cases.décès],
              },
            ],
          }}
          options={{
            legend: { display: false },
            title: { display: true, text: "Nombre de cas confirmés et décès du coronavirus (COVID-19)" }
          }}
        />
      )}
    </>
  );
};
export default BarChart;
