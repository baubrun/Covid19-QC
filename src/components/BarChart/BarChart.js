import React from "react";
import { Bar } from "react-chartjs-2";
import "./BarCart.css";
import PropTypes from "prop-types";

const BarChart = ({ data }) => {
  console.log(data)
  return (
    <>
      {data.length < 1 ? ""
        : (
        <div className="BarChart">
          <Bar
            data={{
              labels: ["Confirmés", "Décès"],
              datasets: [
                {
                  label: "cas",
                  backgroundColor: [
                    "rgba(35, 87, 137, 0.7)",
                    "rgba(193, 41, 46, 0.8)",
                  ],
                  data: [data[0].confirmés, data[0].décès],
                },
              ],
            }}
            options={{
              legend: { display: false },
              responsive: true,
              title: {
                display: true,
              },
            }}
            canvas
          />
        </div>
        )
// {/* )} */}
       
          }
    </>
  );
};

BarChart.propTypes = {
  data: PropTypes.array,
  region: PropTypes.string,
};

export default BarChart;
