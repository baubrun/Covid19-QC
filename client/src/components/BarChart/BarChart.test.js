import React from "react";
import { shallow } from "enzyme";
import BarChart from "./BarChart";
import { checkProps } from "../../../Utils";
import toJson from "enzyme-to-json";

const props = {
  data: [
    {
      confirmés: 48,
      date: "25 mai 2020",
      décès: 2,
      région: "Bas-Saint-Laurent",
    }
  ]
};


describe("BarChart Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<BarChart {...props} />)
  });

  it("should match snapshot ", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });



  it("should render NO errors", () => {
    const component = wrapper.find(".BarChart");
    expect(component.length).toBe(1);
  });

  describe("Checking props", () => {
    it("should NOT throw error", () => {
      const propsError = checkProps(wrapper, props);
      expect(propsError).toBeUndefined();
    });
  });





});
