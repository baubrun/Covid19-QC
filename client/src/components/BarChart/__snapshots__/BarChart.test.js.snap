// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart Component should match snapshot  1`] = `
<Fragment>
  <div
    className="BarChart"
  >
    <Bar
      canvas={true}
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": Array [
                "rgba(35, 87, 137, 0.7)",
                "rgba(193, 41, 46, 0.8)",
              ],
              "data": Array [
                48,
                2,
              ],
              "label": "cas",
            },
          ],
          "labels": Array [
            "Confirmés",
            "Décès",
          ],
        }
      }
      options={
        Object {
          "legend": Object {
            "display": false,
          },
          "responsive": true,
          "title": Object {
            "display": true,
          },
        }
      }
    />
  </div>
</Fragment>
`;
